rustls/src/verify.rs

```diff
@@ -321,6 +321,59 @@ impl fmt::Debug for dyn ClientCertVerifier {
     }
 }
 
+/// Verify that the end-entity certificate `end_entity` is valid and chains to at least one of the [OwnedTrustAnchor] in the `roots` [RootCertStore].
+///
+/// `intermediates` contains all certificates other than `end_entity` that
+/// were sent as part of the server's [Certificate] message. It is in the
+/// same order that the server sent them and may be empty.
+#[allow(unreachable_pub, dead_code)]
+#[cfg_attr(docsrs, doc(cfg(feature = "dangerous_configuration")))]
+pub fn verify_signed_by_trust_anchor(
```

Comment:
I think this name/comment might be masking one important gotcha: `webpki::EndEntity.verify_is_valid_tls_server_cert` also asserts the cert has the `id-kp-serverAuth` extended key usage. E.g. this function might not be what you want if you were trying to verify that a client certificate was signed by a particular trust anchor.

I think at minimum we should call that out in the docstring, but it might be prudent to name this something like `verify_server_cert_signed_by_trust_anchor`.